language/move-lang/tests/stdlib_sanity_check.rs

```diff
@@ -1,66 +1,34 @@
 // Copyright (c) The Diem Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use move_lang::{command_line::read_bool_env_var, move_compile, shared::Address};
+use move_lang::{move_compile, shared::Address};
 use move_lang_test_utils::*;
-use std::{fs, path::Path};
 
-const OUT_EXT: &str = "out";
-
-const KEEP_TMP: &str = "KEEP";
-
-// Runs all tests under the test/testsuite directory.
-fn sanity_check_testsuite_impl(
-    path: &Path,
-    std_lib_is_target: bool,
-    std_lib_dir: String,
-) -> datatest_stable::Result<()> {
-    let mut targets: Vec<String> = vec![path.to_str().unwrap().to_owned()];
-    let mut deps: Vec<String> = vec![];
-    if std_lib_is_target {
-        targets.push(std_lib_dir)
-    } else {
-        deps.push(std_lib_dir)
-    }
+fn sanity_check_testsuite_impl(targets: Vec<String>, deps: Vec<String>) {
     let sender = Some(Address::DIEM_CORE);
 
-    let out_path = path.with_extension(OUT_EXT);
-
-    let (files, units_or_errors) = move_compile(&targets, &deps, sender, None, false)?;
+    let (files, units_or_errors) = move_compile(&targets, &deps, sender, None, false).unwrap();
     let errors = match units_or_errors {
         Err(errors) => errors,
         Ok(units) => move_lang::compiled_unit::verify_units(units).1,
     };
-    let has_errors = !errors.is_empty();
-    let error_buffer = if has_errors {
-        move_lang::errors::report_errors_to_buffer(files, errors)
-    } else {
-        vec![]
-    };
-
-    let save_errors = read_bool_env_var(KEEP_TMP);
 
-    fs::write(out_path.clone(), error_buffer)?;
-    let rendered_errors = fs::read_to_string(out_path.clone())?;
-    if !save_errors {
-        fs::remove_file(out_path)?;
-    }
+    if !errors.is_empty() {
+        let rendered_errors =
+            String::from_utf8(move_lang::errors::report_errors_to_buffer(files, errors)).unwrap();
 
-    if has_errors {
-        let msg = format!("Expected success. Unexpected errors:\n{}", rendered_errors);
-        error(msg)
-    } else {
-        Ok(())
+        panic!("Expected success. Unexpected errors:\n{}", rendered_errors);
```

Comment:
Not clear why you're changing this from "error(...)" to panic. I'm not saying it's wrong, but I'm at least curious....

Reply:
TLDR: this is because `fn error` returns a `datatest_stable::Result<()>` and since we're no longer using datatest I changed it to a simple panic.

However you just made me realize that I should check if `fn error` and a bunch of other things in the same crate still have any clients, and clean them up if the answer is no.